src/modules/flight_mode_manager/FlightModeManager.cpp

```diff
@@ -146,23 +145,7 @@ void FlightModeManager::start_flight_task()
 
 	// Only run transition flight task if altitude control is enabled (e.g. in Altitdue, Position, Auto flight mode)
 	if (_vehicle_status_sub.get().in_transition_mode && _vehicle_control_mode_sub.get().flag_control_altitude_enabled) {
-
-		should_disable_task = false;
-		FlightTaskError error = switchTask(FlightTaskIndex::Transition);
-
-		if (error != FlightTaskError::NoError) {
-			if (prev_failure_count == 0) {
-				PX4_WARN("Transition activation failed with error: %s", errorToString(error));
-			}
-
-			task_failure = true;
-			_task_failure_count++;
-
-		} else {
-			// we want to be in this mode, reset the failure count
-			_task_failure_count = 0;
-		}
-
+		switchTask(FlightTaskIndex::Transition);
```

Comment:
I noticed the transition flight task cannot have a failure anymore with this mechanism. I don't know if that's a problem. But at least it's different from before and the error handling of starting the other tasks.

Reply:
Both `activate` and `updateInitialize` for the transition always return true, do I miss something?

Reply:
No in most cases e.g. transition they do. The mechanism is thought to handle recognizable execution errors but we can solve that differently anyways 👍🏻